dotnet-desktop-guide/net/winforms/printing/snippets/how-to-print-text-document/csharp/Form1.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace Sample_Print_Application
+{
+    public partial class Form1 : Form
+    {
+        //<add_string_to_your_form>
+        private string stringToPrint="";
+        //</add_string_to_your_form>
+
+        public Form1()
+        {
+            InitializeComponent();
+        }
+        
+        //<print_contents_using_event_handler>
+        private void PrintDocument1_PrintPage(object sender, System.Drawing.Printing.PrintPageEventArgs e)
+        {
+            int charactersOnPage = 0;
+            int linesPerPage = 0;
+
+            // Sets the value of charactersOnPage to the number of characters
+            // of stringToPrint that will fit within the bounds of the page.
+            e.Graphics.MeasureString(stringToPrint, this.Font,
+                e.MarginBounds.Size, StringFormat.GenericTypographic,
+                out charactersOnPage, out linesPerPage);
+
+            // Draws the string within the bounds of the page
+            e.Graphics.DrawString(stringToPrint, this.Font, Brushes.Black,
+                e.MarginBounds, StringFormat.GenericTypographic);
+
+            // Remove the portion of the string that has been printed.
+            stringToPrint = stringToPrint.Substring(charactersOnPage);
+
+            // Check to see if more pages are to be printed.
+            e.HasMorePages = (stringToPrint.Length > 0);
+        }
+        //</print_contents_using_event_handler>
+        
+        private void button1_Click(object sender, EventArgs e)
+        {
+            //<set_DocumentName_and_string>
+            string docName = "testPage.txt";
+            string docPath = @"C:\";
+            printDocument1.DocumentName = docName;
+            using FileStream stream = new (docPath + docName, FileMode.Open);
+            using StreamReader reader = new (stream);
+            {
+                stringToPrint = reader.ReadToEnd();
+            }
+            
+            //<call_print_method_to_print_file>
+            printDocument1.Print();
+            //</call_print_method_to_print_file>
+
+            //</set_DocumentName_and_string>
+
+        }
```

Comment:
This code can be tightened up a bit. Here are the changes I did:

1. I moved the code block ID to outside the button click, so that it includes the button click code and people can see where the code is running from.
2. I added the `fullPath` variable to the code which uses the `Path.Combine` method to join the two paths instead of using a `+` operator
3. Instead of a stream reader, I used the `File.ReadAllText` method which does the same thing in a single line.

```suggestion
        //<set_DocumentName_and_string>
        private void button1_Click(object sender, EventArgs e)
        {
            string docName = "testPage.txt";
            string docPath = @"C:\";
            string fullPath = System.IO.Path.Combine(docPath, docName);

            printDocument1.DocumentName = docName;

            stringToPrint = System.IO.File.ReadAllText(fullPath);
            
            //<call_print_method_to_print_file>
            printDocument1.Print();
            //</call_print_method_to_print_file>
        }
        //</set_DocumentName_and_string>
```